### DISCUSSION TITLE
Dynamic orderby

### DISCUSSION THREAD

**Discussion by darylhjd** [go-jet/jet#265] [Q&A]
Is there a way to construct dynamic order by using strings from, example, a http query parameter?

For example, if a request wants to sort by a column `id` for table `Users`, then we can use this string to get the `Users.ID` column.

Since I also wanted to reuse this functionality across different tables, I also tried implementing this function:
```go
func getColumn(table Table, colName string) (Column, error) {
	for _, col := table.columns() {
		if col.Name == colName {
			return col, nil
		}
	}

	return nil, errors.New("no matched columns")
}
```

but the `columns()` method is unfortunately not exported for public usage. Just wondering what is the general practice for such use cases. Thanks!

**Comment by darylhjd** [comment]
my suggestion, if possible, would be to make `columns` public, or perhaps even introduce a `GetColumn(s string)` method to the `Table` for general usage.

**Comment by go-jet** [comment]
You can use `Table.AllColumns` slice:

```golang
func getColumn(columns ColumnList, colName string) (Column, error) {
	for _, col := range columns {
		if col.Name() == colName {
			return col, nil
		}
	}

	return nil, errors.New("no matched columns")
}


col, err := getColumn(Actor.AllColumns, "first_name")


```

**Comment by go-jet** [comment]
Maybe we can add `Column(name string) (Column, error)` method on `ColumnList` type.
